dasch-swiss/dsp-app
# feat(error): Server error handler (DSP-710)

resolves DSP-710 (s. discussion)

closes #311 

- Implementation of error handler service with following job:
  - depending on error status show message from dsp-ui notification service (mat snackbar) or show full-size error page (case 500 and 503 / server error)

- Add error handler to all components and services where we do api requests (incl. refactoring of Imports)
- Error test page `/teapot` with status 418 ([easteregg](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/418))
  - shows the default app error; default means it's not one of the most common error we have: 403, 404, 500 or 503
  - change the status number in `app-routing.module.ts` from 418 to another one (e.g. 403, 404, 500 or 503) to see the different error pages
 
- Test the implementation of error handler:
  - run the app and turn off knora-stack (error 500) or at least fuseki (error 503): it should show a full-size "Service unavailable" error message
  - turn on knora-stack or fuseki again and press "reload page" link in the error message and it should load the page again
  - with turned-off fuseki the only page that works is the `/help` page because there we do not do db-relevant requests.

Comment: We could add a link to <https://status.dasch.swiss> in one of the error 5xx page. So, user will see what's going on on the server. At the moment this service is only implemented for the prod server and not yet for the test (or staging) server which would be <https://status.test.dasch.swiss> (<https://status.staging.dasch.swiss>). Btw with this status page the ops-developer team will be informed when a service doesn't run anymore. So, we don't need a contact button anymore on the server error page to send this kind of error report.
